apps/wing-console/console/ui/src/services/use-explorer.tsx

```diff
@@ -56,15 +57,17 @@ export const useExplorer = () => {
   );
 
   useEffect(() => {
-    if (!tree.data) {
+    if (!tree.data || !nodeIds.data) {
       return;
     }
     setItems([createTreeMenuItemFromExplorerTreeItem(tree.data)]);
 
-    setSelectedNode.mutate({
-      resourcePath: "root",
-    });
-  }, [tree.data, setItems]);
+    if (!selectedNode.data || !nodeIds.data?.includes(selectedNode.data)) {
+      setSelectedNode.mutate({
+        resourcePath: "root",
+      });
+    }
+  }, [tree.data, setItems, nodeIds.data]);
```

Comment:
I feel like this should become two different effects. Otherwise, `setItems` and `createTreeMenuItemFromExplorerTreeItem` will be called more than once if I'm not mistaken.